lib/common/client/middleware/response/mhv_xml_html_errors.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+module Common
+  module Client
+    module Middleware
+      module Response
+        class MhvXmlHtmlErrors < Faraday::Response::Middleware
+          ERROR_LIST = %w(service_outage generic_xml generic).freeze
+
+          def on_complete(env)
+            return if ok_or_json?(env)
+
+            @doc = html?(env) ? Nokogiri::HTML(env.body) : Nokogiri::XML(env.body)
+
+            jsoned_error = ERROR_LIST.each_with_object('') { |error, je| break je if (je = send(error)) }
+            jsoned_error['errorCode'] = env.status if jsoned_error['errorCode'].blank?
+
+            env.body = jsoned_error
+            env.response_headers['content-type'] = 'application/json'
+          end
+
+          private
+
+          attr_reader :doc
+
+          def ok_or_json?(env)
+            [1, 2, 3].include?(env.status / 100) || env.response_headers['content-type'] =~ /\bjson/
+          end
+
+          def html?(env)
+            env.response_headers['content-type'] =~ /html/i
+          end
+
+          def service_outage
+            fault = doc.xpath('//errormsg:Fault', 'errormsg' => 'http://schemas.xmlsoap.org/soap/envelope/')
+            fault_code = fault.at_css('faultcode')
```

Comment:
Simply to verify structure - that this is exactly the error we are getting, otherwise its handled by a more generic method. But I can remove it easy enough.